### PR TITLE
Replaced gzip with lzma

### PR DESCRIPTION
**Pull request by KOLANICH**

### CODE REVIEW

**Comment by pietermarsman** [comment]
Hi @kolanich, thanks for your suggestions and improvement proposals. 

Is there any particular reason you need to have better compression? Otherwise, I think we should just leave it the way it is.

**Comment by KOLANICH** [comment]
>Is there any particular reason you need to have better compression? Otherwise, I think we should just leave it the way it is.

Free space on disk is always an issue, even if it is a few KiB, so as long as performance is satisfactory, the algo compressing the best should be used. IMHO performance of decompression of LZMA and LZMA2 is satisfactory.

**Comment by pietermarsman** [comment]
I think disk issue isn't a major concern for most users of pdfminer. I suggest we focus on your other PR's and that you close this one.

File: pdfminer/cmapdb.py [hunk 1]
```diff
@@ -231,14 +231,14 @@ class CMapNotFound(CMapError):
     @classmethod
     def _load_data(klass, name):
         name = name.replace("\0", "")
-        filename = '%s.pickle.gz' % name
+        filename = '%s.pickle.xz' % name
```
Comment (pietermarsman) [inline pdfminer/cmapdb.py]: The cmap files are included into the package (in the `pdfminer/cmp` directory). Changing the extension invalidates the path.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Contributing guidelines
```
Comment (pietermarsman) [inline CONTRIBUTING.md]: These files are from another PR. Could you remove all the changes that are not part of this PR?

File: pdfminer/cmapdb.py [hunk 3]
```diff
@@ -231,14 +231,14 @@ class CMapNotFound(CMapError):
...
         log.info('loading: %r', name)
         cmap_paths = (os.environ.get('CMAP_PATH', '/usr/share/pdfminer/'),
                       os.path.join(os.path.dirname(__file__), 'cmap'),)
         for directory in cmap_paths:
             path = os.path.join(directory, filename)
             if os.path.exists(path):
-                gzfile = gzip.open(path)
+                gzfile = lzma.open(path)
```
Comment (pietermarsman) [inline pdfminer/cmapdb.py]: The cmap files are included into the package (in the `pdfminer/cmp` directory). By changing the compression method those files cannot be used anymore.

File: CONTRIBUTING.md (hunk 2 above)
Comment (KOLANICH) [reply CONTRIBUTING.md]: I have rebased over `devel`. Probably I should have sent the PR there.

**Comment by pietermarsman** [comment]
I'm closing this issue because I think there is no need for better file compression just now. If you are having specific problems due to pdfminer.six being to large, please create an issue about that.
